src/preamble.js

```diff
@@ -779,12 +779,9 @@ function demangle(func) {
   var __cxa_demangle_func = Module['___cxa_demangle'] || Module['__cxa_demangle'];
   assert(__cxa_demangle_func);
   try {
-    var s =
-#if WASM_BACKEND
-      func;
-#else
-      func.substr(1);
-#endif
+    var s = func;
+    if (s.startsWith('__Z'))
```

Comment:
It does split it off, if there's an extra one. C++ starts with `_Z`, so if it has two `_`s it's because asm.js added another one.
And this calls cxa_demangle, so I assumed it was C++. Do we ever call demangle on C functions that get linked in with C++ ones? If so that might be confusing.

I'm also not sure why this change was made, it turns a compile-time thing into a runtime thing. Statically linking fastcomp and upstream wasm modules?

Reply:
Oh right, reading the calling function, we do differentiate the two backends there, and indeed the difference is the same as checking if it starts with `__Z` or `_Z`.

So this part lgtm with the extra `{`, `}` (or the body on the same line).

Reply:
The problem here is really the code in `test_demangle` which assumes the double underscore in all cases.   We might want to update that.